JASP-Desktop/widgets/qmllistview.cpp

```diff
@@ -63,6 +64,51 @@ QList<QVariant> QMLListView::_getListVariant(QVariant var)
 	return listVar;
 }
 
+QString QMLListView::_readSourceName(const QString& sourceNameExt, QString& sourceControl, QString& sourceUse)
+{
+	QStringList nameSplit = sourceNameExt.split(".");
+	if (nameSplit.length() > 1)
+	{
+		sourceControl = nameSplit[1];
+		sourceUse = "control=" + nameSplit[1];
+	}
+
+	return nameSplit[0];
+}
+
+QMap<QString, QVariant> QMLListView::_readSource(const QVariant& source, QString& sourceName, QString& sourceControl, QString& sourceUse)
+{
+	QMap<QString, QVariant> map;
+
+	JASPControlBase* sourceItem = source.value<JASPControlBase*>();
+	if (sourceItem)
+		sourceName = sourceItem->name();
+	else if (source.canConvert<QString>())
```

Comment:
So and int in a QVariant *can* be converted to a QString, wouldn't it be safer to check for ["type()"](https://doc.qt.io/qt-5/qvariant.html#type) ?

Reply:
Or you always want it to just be converted to a string unless it is a map? Because in that case I would put the map first, it seems like it would be a more restrictive "canConvert"

Reply:
Maybe Qt may decide one day that a property value like '{"name": "test" }' can be converted to a string... you never know... OK, I change the order of the if else.

Reply:
Wouldn't it be better to just check the type instead?